src/core/public/chrome/utils.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { AppCategory } from 'opensearch-dashboards/public';
+import { groupBy, sortBy } from 'lodash';
+import { ChromeNavLink } from './nav_links';
+import { ChromeRegistrationNavLink } from './nav_group';
+
+export const LinkItemType = {
+  LINK: 'link',
+  CATEGORY: 'category',
+  PARENT_LINK: 'parentLink',
+} as const;
+
+export type LinkItem = { order?: number } & (
+  | { itemType: 'link'; link: ChromeNavLink & ChromeRegistrationNavLink }
+  | { itemType: 'parentLink'; link?: ChromeNavLink & ChromeRegistrationNavLink; links: LinkItem[] }
+  | { itemType: 'category'; category?: AppCategory; links?: LinkItem[] }
+);
+
+export function getAllCategories(allCategorizedLinks: Record<string, ChromeNavLink[]>) {
+  const allCategories = {} as Record<string, AppCategory | undefined>;
+
+  for (const [key, value] of Object.entries(allCategorizedLinks)) {
+    allCategories[key] = value[0].category;
+  }
+
+  return allCategories;
+}
+
+export function fulfillRegistrationLinksToChromeNavLinks(
```

Comment:
Would you mind to leave a comment to describe this function?

Reply:
Sure, done for that.